### PR TITLE
Ton of doctests added

### PR DESCRIPTION
**Pull request by arcAman07**
Added a ton of doctests to index.md ( at the moment ). Will add doctests to all the markdown files if it needs to be added.
This will help to avoid documentation examples from becoming outdated, incorrect, or misleading.

### CODE REVIEW

**Comment by arcAman07** [comment]
Added a couple of relevant doctests to the index.md folder, but am not able to figure out the " 1 failing check" during this PR. Can somebody please help me with this. ( as all tests are running properly)

**Comment by ToucheSir** [comment]
Seems like a failure on nightly and unrelated.

**Comment by arcAman07** [comment]
> Seems like a failure on nightly and unrelated.

Ohk 🙃, I couldn't actually understand the reason as both the doctests and tests were working mine on my pc. Would love some feedbacks over the doctests , planning to add more to the other markdown files. Omitted doctests on code snippets where we dealing with randomized values with rand() function

**Comment by CarloLucibello** [comment]
hi @arcAman07 this is a welcome contribution, but as usual I ask you to not create a ton of commits besides the ton of doctests. Take your time to do all the changes you plan to do then push only when you fill the PR is ready or you need some feedback. Also, build the documentation locally and see that you don't get errors.

File: docs/make.jl [hunk 1]
```diff
@@ -8,6 +8,7 @@ using Documenter, Zygote
 
 makedocs(
   sitename="Zygote",
+  doctest = false,
```
Comment (CarloLucibello) [inline docs/make.jl:11]: ?

**Comment by arcAman07** [comment]
> hi @arcAman07 this is a welcome contribution, but as usual I ask you to not create a ton of commits besides the ton of doctests. Take your time to do all the changes you plan to do locally, then push only when you fill the PR is ready or you need some feedback. Also, build the documentation locally and see that you don't get errors.

Yeah really my bad, I actually was running them locally by doctests =  false , and I kinda forgot I was using this package which I had created which was commiting the changes after each change. Just disabled that package currently, will be more careful 👍

File: docs/make.jl (hunk 1 above)
Comment (arcAman07) [reply docs/make.jl]: Changed made , was testing locally.

**Comment by arcAman07** [comment]
Added the necessary doctests,  would love some improvements that can be made to the doctests. Just the nightly isn't running properly.

**Comment by CarloLucibello** [comment]
Thanks!

**Pull request merged**